.github/actions/node-cache/action.yml

```diff
@@ -8,14 +8,20 @@ runs:
       id: tooling-versions
       shell: bash
       run: |
-        echo "node=$(./mvnw help:evaluate -Dexpression=node.version -q -DforceStdout | cut -c 2-)" >> $GITHUB_OUTPUT
-        echo "pnpm=$(./mvnw help:evaluate -Dexpression=pnpm.version -q -DforceStdout)" >> $GITHUB_OUTPUT
+        echo "node=$(cat pom.xml | grep '<node.version>' | cut -d '>' -f 2 | cut -d '<' -f 1 | cut -c 2-)" >> $GITHUB_OUTPUT
+        echo "pnpm=$(cat pom.xml | grep '<pnpm.version>' | cut -d '>' -f 2 | cut -d '<' -f 1 | cut -c 1-)" >> $GITHUB_OUTPUT
```

Comment:
This is much faster, but also will break:

1. if the XML syntax isn't simple, or
2. if the property is overridden within the maven pom inheritance

Something that's equally fast and less susceptible to 1 is the following:

```bash
sed -r 's,^.*<pnpm.version>([^<]+)</pnpm.version>.*$,\1,;tx;d;:x' pom.xml
```

or this which needs an extra package installed, but is totally immune to 1:

```
xmlstarlet sel -t -m '//_:project/_:properties' -v '_:node.version' -n pom.xml 
```

Reply:
I honestly prefer using `grep` and `cut` here as it is much easier to follow, even if it is more verbose. We're always parsing the same POM file in the root, so we don't have to worry about things like inheritance. The XML is also a simple property definition, so it won't be a problem to parse, even in the future.

I did consider using [`xq`](https://github.com/sibprogrammer/xq?tab=readme-ov-file#xq) here, which is basically `jq` for XML, but it would involve installing another package.